kasa/discover.py

```diff
@@ -743,6 +743,8 @@ def _get_device_instance_legacy(info: dict, config: DeviceConfig) -> Device:
         device = device_class(config.host, config=config)
         sys_info = _extract_sys_info(info)
         device_type = sys_info.get("mic_type", sys_info.get("type"))
+        if device_type is None:
+            raise UnsupportedDeviceError("type nor mic_type found in sysinfo response")
```

Comment:
Unsure why this was not spotted earlier by mypy, but `from_values` below expects that this is not none.